relay_client/src/http.rs

```diff
@@ -0,0 +1,281 @@
+        })
+    }
+
+    /// Publishes a message over the network on given topic.
+    pub async fn publish(
+        &self,
+        topic: Topic,
+        message: impl Into<Arc<str>>,
+        tag: u32,
+        ttl: Duration,
+    ) -> EmptyResponse {
+        self.request(rpc::Publish {
+            topic,
+            message: message.into(),
+            ttl_secs: ttl.as_secs() as u32,
+            tag,
+            prompt: false,
+        })
+        .await
+        .map(|_| ())
+    }
+
+    /// Subscribes on topic to receive messages.
+    pub async fn subscribe(&self, topic: Topic) -> Response<rpc::Subscribe> {
+        self.request(rpc::Subscribe { topic }).await
+    }
+
+    /// Unsubscribes from a topic.
+    pub async fn unsubscribe(
+        &self,
+        topic: Topic,
+        subscription_id: SubscriptionId,
+    ) -> Response<rpc::Unsubscribe> {
+        self.request(rpc::Unsubscribe {
+            topic,
+            subscription_id,
+        })
+        .await
+    }
+
+    /// Fetch mailbox messages for a specific topic.
+    pub async fn fetch(&self, topic: Topic) -> Response<rpc::FetchMessages> {
+        self.request(rpc::FetchMessages { topic }).await
+    }
+
+    /// Registers a webhook to watch messages.
+    pub async fn watch_register(
+        &self,
+        request: WatchRegisterRequest,
+        keypair: &Keypair,
+    ) -> Response<rpc::WatchRegister> {
+        let iat = chrono::Utc::now().timestamp();
+        let ttl_sec: i64 = request
+            .ttl
+            .as_secs()
+            .try_into()
+            .map_err(|err| HttpClientError::InvalidRequest(Box::new(err)))?;
+        let exp = iat + ttl_sec;
+
+        let claims = rpc::WatchRegisterClaims {
+            basic: jwt::JwtBasicClaims {
+                iss: DecodedClientId::from_key(&keypair.public_key()).into(),
+                aud: self.origin.clone(),
+                iat,
+                sub: request.service_url,
+                exp: Some(exp),
+            },
+            act: rpc::WatchAction::Register,
+            typ: request.watch_type,
+            whu: request.webhook_url,
+            tag: request.tags,
+            sts: request.statuses,
+        };
+
+        let payload = rpc::WatchRegister {
+            register_auth: claims.encode(keypair).map_err(HttpClientError::Jwt)?,
+        };
+
+        self.request(payload).await
+    }
+
+    /// Unregisters a webhook to watch messages.
+    pub async fn watch_unregister(
+        &self,
+        request: WatchUnregisterRequest,
+        keypair: &Keypair,
+    ) -> Response<rpc::WatchUnregister> {
+        let iat = chrono::Utc::now().timestamp();
+
+        let claims = rpc::WatchUnregisterClaims {
+            basic: jwt::JwtBasicClaims {
+                iss: DecodedClientId::from_key(&keypair.public_key()).into(),
+                aud: self.origin.clone(),
+                iat,
+                sub: request.service_url,
+                exp: Some(iat + 60 * 60),
```

Comment:
Shouldn't we rely on some constant instead of a magic number?

Reply:
There should actually be no `exp` at all.